core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/AssetIndexLocator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (c) 2021 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial API and Implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.contract;
+
+import org.eclipse.dataspaceconnector.spi.asset.AssetIndex;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+class AssetIndexLocator {
```

Comment:
Maybe create a registry, where asset indices are registered. This way its the same solution, as in the other extensions. And create an issue, that it should be possible to register multiple instances of the same type.